### PR TITLE
Easi 3288/discussion tagging email

### PR DESCRIPTION
**Pull request by StevenWadeOddball**
# EASI-3288
## Changes and Description

- Store a reference to the tagged user or possible solution in the HTMLMention struct ( so you don't have to query the database again)
- Email when a user is tagged in a discussion or in a reply
- Email to Solution POCs when a possible solution is tagged in a discussion or a in a reply
   - Added an email config so that emails to POCs can be turned on and off by environment (so that real pocs aren't emails in LLES)
   

## How to test this change
1.  Use the postman collection to create a model plan
2. Use the collection to make a discussion
3. Look in [mailcatcher](http://127.0.0.1:1085/) to verify that you received an email for every unique tagged user / solution
4. Use postman to make a reply.
5. Look in [mailcatcher](http://127.0.0.1:1085/) to verify that you received an email for every unique tagged user / solution 
6. Verify that the emails match the design specifications from [FIGMA](https://www.figma.com/file/8MfXzXzmKWv8vBjTasUIkD/MINT---Notifications?type=design&node-id=3079-3721&mode=design&t=yT9qNTGBUr4qdDgG-0)

If desired, you can also test this using the frontend to make tags, and verify that emails are sent as expected

Also, to look at just the emails, you can run `scripts/dev email` to see the new emails.

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [x] Added or updated tests for backend resolvers or other functions as needed.
- [x] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by patrickseguraoddball — APPROVED** [review]
Looks and works great and as expected!  Thanks!

**Comment by StevenWadeOddball** [comment]
Thanks @patrickseguraoddball! 
@ClayBenson94 , @OddTomBrooks , I implemented the option we were talking about when send poc emails is turned off. Now it will send to the dev email group with the name of the POC appended to the email.

File: pkg/graph/resolvers/plan_discussion.go [hunk 1]
```diff
@@ -133,10 +343,11 @@ func sendPlanDiscussionCreatedEmail(
 	emailBody, err := emailTemplate.GetExecutedBody(email.PlanDiscussionCreatedBodyContent{
 		ClientAddress:     emailService.GetConfig().GetClientAddress(),
 		DiscussionID:      planDiscussion.ID.String(),
-		UserName:          planDiscussion.CreatedByUserAccount(ctx).CommonName,
+		UserName:          createdByUserName,
 		DiscussionContent: planDiscussion.Content.RawContent.ToTemplate(),
 		ModelID:           modelPlan.ID.String(),
 		ModelName:         modelPlan.ModelName,
+		Role:              createdByUserRole, //TODO verify with Natasha
```
Comment (ClayBenson94) [inline pkg/graph/resolvers/plan_discussion.go:350]: Is this still `TODO`?

**Review by ClayBenson94 — DISMISSED** [review]
No major review comments -- just a few `TODO` comments we could probably clean up, and that one about the variable!

File: pkg/graph/resolvers/plan_discussion.go (hunk 1 above)
Comment (StevenWadeOddball) [reply pkg/graph/resolvers/plan_discussion.go:350]: Thanks! No, I verified about this, thanks for catching it!

File: pkg/graph/resolvers/plan_discussion.go [hunk 2]
```diff
@@ -93,20 +101,226 @@ func CreatePlanDiscussion(
...
+			store,
+			logger,
+			emailService,
+			emailTemplateService,
+			addressBook,
+			discussion.Content,
+			discussion.ID,
+			modelPlan,
+			commonName,
+			discussion.UserRole.Humanize(models.ValueOrEmpty(discussion.UserRoleDescription)))
```
[the remark this answers is not in the document]
Comment (StevenWadeOddball) [reply pkg/graph/resolvers/plan_discussion.go]: Good call! Thanks Clay, I've updated it

**Pull request merged**